pandas/tests/tools/test_merge.py

```diff
@@ -1355,3 +1355,92 @@ def test_dtype_on_merged_different(self, change, how, left, right):
                            np.dtype('int64')],
                           index=['X', 'Y', 'Z'])
         assert_series_equal(result, expected)
+
+
+class TestMergeOnIndexes(object):
+
+    def test_merge_on_indexes(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
```

Comment:
I would change this to be parametrized like join above. (e.g. matrix of how/sort)